src/applications/appeals/10182/containers/FormApp.jsx

```diff
@@ -109,6 +103,17 @@ export const FormApp = ({
     [loggedIn, contestableIssues, showPart3, formData.contestedIssues],
   );
 
+  useEffect(
+    () => {
+      setFormData({
+        ...formData,
+        [SHOW_PART3]: showPart3,
+      });
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [showPart3, formData[SHOW_PART3]],
```

Comment:
just a gut check — is there any chance that this could infinitely loop because `formData[SHOW_PART3]` could continuously get updated? i can't remember if the dependency check looks at the actual value or if it relies on if there has generally been a state change 🤔

Reply:
That's a great question @anniebtran. I tested locally and didn't experience an infinite loop, but this code reads as if it should trigger an infinite loop. I'm not sure why it isn't. @Mottie, what's the `formData[SHOW_PART3]` for here?

Reply:
Let me know what you think, but I think the only dependency we care about here is `showPart3`

Reply:
The `formData[SHOW_PART3]` dependency was added because the form save-in-progress was occurring _after_ this code was executed causing the feature toggle value within the form data to not be in sync with the actual feature toggle setting. I can see how this could potentially cause an infinitely loop because I can't recall, or seem to find, if the values in the dependency array are only triggered from a value change or a value update.